### PR TITLE
[Multi-Tab] PSEUDOCODE

### PR DESCRIPTION
**Pull request by schmidt-sebastian**

### CODE REVIEW

File: packages/firestore/src/local/tab_notification_channel.ts [hunk 1]
```diff
@@ -102,6 +171,81 @@ export class LocalStorageNotificationChannel implements TabNotificationChannel {
...
+    this.rejectedTargetIds[targetId] = { date: new Date(), error };
+  }
+
+  updateQuery(updatedTargetIds: TargetId[]): void {
+    for (const targetId of updatedTargetIds) {
+      this.updatedTargetIds[targetId] = new Date();
+    }
+  }
+
+  private onUpdate(key: string, value: string) {
```
Comment (mikelehen) [inline packages/firestore/src/local/tab_notification_channel.ts]: Your pseudocode needs more comments and less code. :-) Who calls this? when?

File: packages/firestore/src/local/tab_notification_channel.ts [hunk 2]
```diff
@@ -102,6 +171,81 @@ export class LocalStorageNotificationChannel implements TabNotificationChannel {
     this.persistState();
   }
 
+  addMutation(batchId: BatchId): void {
+    this.pendingBatchIds[batchId] = this.instanceId;
```
Comment (mikelehen) [inline packages/firestore/src/local/tab_notification_channel.ts]: I assume the intention is that these write to local storage? Again, more comments and less code would be more helpful.

File: packages/firestore/src/local/tab_notification_channel.ts (hunk 1 above)
Comment (schmidt-sebastian) [reply packages/firestore/src/local/tab_notification_channel.ts]: All these methods are supposed to call this.persistState(). Comment added.

Comment (schmidt-sebastian) [reply packages/firestore/src/local/tab_notification_channel.ts]: This is the event listener hook called when a key in LocalStorage changes.

**Pull request closed without merging**